src/components/cylc/Graph.vue

```diff
@@ -0,0 +1,1644 @@
+            edgeObj.data.name = this.workflowName
+          }
+          has(edge, 'label') && !isEmpty(edge.label) ? edgeObj.data.label = edge.label : edgeObj.label = ''
+          edgeObj.data.source !== undefined || edgeObj.data.target !== undefined ? edgesArray.push(edgeObj)
+            : console.debug('skipping adding edge with empty source or target')
+        })
+        // console.debug('EDGES ::: ', JSON.stringify(edgesArray))
+        return edgesArray
+      } catch (error) {
+        console.error('getEdges error: ', error)
+      }
+    },
+
+    getJobs (jobs) {
+      const jobsArray = []
+      each(jobs, (job, key) => {
+        jobsArray.push(job)
+      })
+      return jobsArray
+    },
+
+    changeLayout (value) {
+      try {
+        if (this.freeze) {
+          return
+        }
+        if (!isUndefined(value) && !isEmpty(value)) {
+          this.layoutName = value
+        }
+      } catch (error) {
+        console.error('changeLayout error: ', error)
+      }
+    },
+
+    async handleMounted () {
+      try {
+        this.status = 'success'
+      } catch (error) {
+        console.error('handleMounted error: ', error)
+        this.status = 'error'
+      }
+    },
+
+    async preConfig (cytoscape) {
+      // cytoscape: this is the cytoscape constructor
+      try {
+        console.debug('PRE-CONFIG')
+        cytoscape.use(cola)
+        cytoscape.use(dagre)
+        cytoscape.use(cise)
+        cytoscape.use(coseBilkent)
+        this.cy = cytoscape({
+          container: document.getElementById('cytoscape'),
+          elements: this.graphData
+        })
+        this.debouncer = debounce(this.updateGraph, 100)
+      } catch (error) {
+        console.error('preConfig error: ', error)
+      }
+    },
+
+    async afterCreated (cy) {
+      try {
+      // load graph data and run layout
+        layoutOptions = dagreOptions
+        expandCollapseOptions = expandCollapseOptionsUndefined
+        const loaded = await this.initialise(cy)
+        loaded ? this.loading = false : console.error('there was an error loading the graph view')
+      } catch (error) {
+        console.error('afterCreated error', error)
+      }
+    },
+
+    async registerExtensions (instance) {
+      try {
+        // register extensions
+        if (typeof cytoscape('core', 'navigator') !== 'function') {
+          console.debug('registering navigator')
+          navigator(cytoscape)
+        }
+
+        if (typeof cytoscape('core', 'panzoom') !== 'function') {
+          console.debug('registering panzoom')
+          panzoom(cytoscape)
+        }
+
+        if (typeof cytoscape('core', 'undoRedo') !== 'function') {
+          console.debug('registering undoRedo')
+          undoRedo(cytoscape)
+        }
+
+        if (typeof cytoscape('core', 'expandCollapse') !== 'function') {
+          console.debug('registering expandCollapse with jquery')
+          expandCollapse(cytoscape, jquery)
```

Comment:
Using the IDE, I couldn't locate where the `jquery` value was used.

![image](https://user-images.githubusercontent.com/304786/67732149-51a33b80-fa5f-11e9-81bc-8c3cb39eba54.png)

They also had a PR some time ago to remove the JQuery dependency: https://github.com/iVis-at-Bilkent/cytoscape.js-expand-collapse/pull/75/files

So if that's correct, we could update this method, and also remove the JQuery dependency in the `package.json`. Or, if we decide to move the expand/collapse to a follow-up PR, then both dependency and this block of code could be removed?

Reply:
well spotted, Id removed this before but it crept back!

Reply:
removed the dependency on jquery